File: backend/app/models/scout.py
"""Scout data models for Yutori monitoring tasks."""
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class Scout(BaseModel):
    """A Yutori scouting task."""
    id: str
    brand_id: str
    query: str
    display_name: str
    status: str = "active"
    output_interval: int = 86400
    created_at: datetime
    next_run: Optional[datetime] = None


class ScoutUpdate(BaseModel):
    """An update received from a Yutori scout."""
    id: str
    scout_id: str
    content: str
    citations: list[dict] = []
    created_at: datetime

File: frontend/src/types/scout.ts
/** A Yutori monitoring scout */
export interface Scout {
  id: string;
  brand_id: string;
  query: string;
  display_name: string;
  status: "active" | "paused" | "done";
  output_interval: number;
  created_at: string;
  next_run?: string;
}

/** An update from a scout */
export interface ScoutUpdate {
  id: string;
  scout_id: string;
  content: string;
  citations: Citation[];
  created_at: string;
}

export interface Citation {
  id: string;
  url: string;
  preview_data?: Record<string, unknown>;
}
